input.lua

```diff
@@ -175,6 +175,87 @@ end
 
 function courseplay:executeFunction(self, func, value, page)
 	courseplay:debug("executeFunction: function: " .. func .. " value: " .. tostring(value) .. " page: " .. tostring(page), 5)
+	local validateClasses={
+		"Setting", --cp.settings
+		"CourseGeneratorSetting", --cp.courseGeneratorSettings
+		"AIDriver" --cp.driver
+	}
+	--for now only support for Setting and CourseGeneratorSetting and Driver Classes
+	--needs more optimization!!!
+	local callError = nil
+	local validCall = false
+	sepFunc = {}
+	courseplay:splitFunctionCall(func,sepFunc)
+	for index, call in ipairs(sepFunc) do 
+		courseplay:debug("Index: "..index.." call: "..call,5)
+	end
+	if #sepFunc > 3 then 
+		callError = "to many seperate funcCalls!"
+	end
+	if #sepFunc > 1 and callError==nil then
```

Comment:
See my other comment on remembering the object, not only the function to save us this effort.